parser.py

```diff
@@ -0,0 +1,168 @@
+import scanner
+import ply.yacc as yacc
+import lexer as scanner
+
+
+tokens = scanner.tokens
+
+precedence = (
+   ("nonassoc", 'IF'),
+   ("nonassoc", 'ELSE'),
+   ("nonassoc", '<', '>', 'EQ', 'NOTEQ', 'GE', 'LE'),
+   ("left", '+', '-'),
+   ("left", 'DOTADD', 'DOTSUB'),
+   ("left", '*', '/'),
+   ("left", 'DOTMUL', 'DOTDIV'),
+   ("left", 'TRANSPOSE'),
+   ("right", 'UMINUS')
+)
+
+
+def p_error(p):
+    if p:
+        print("Syntax error at line {0}: LexToken({1}, '{2}')".format(p.lineno, p.type, p.value))
+    else:
+        print("Unexpected end of input")
+
+
+def p_program(p):
+    """program : instructions_opt"""
+
+def p_instructions_opt_1(p):
+    """instructions_opt : instructions """
+
+def p_instructions_opt_2(p):
+    """instructions_opt : """
+
+def p_instructions_1(p):
+    """instructions : instructions instruction """
+
+def p_instructions_2(p):
+    """instructions : instruction """
+
+def p_instruction(p):
+    """ instruction :  if
+                    |  for
+                    |  while
+                    |  break
+                    |  continue
+                    |  print
+                    |  block
+                    |  return 
+                    |  assign 
+                    |  assign_array
+                    """
+
+def p_num(p):
+    """ num : INTNUM
+            | FLOAT """
+
+def p_val(p):
+    """ value : STRING
+              | num """
+
+def p_val_array(p):
+    """ values : value
+               | value ',' values"""
```

Comment:
spróbuj poprawić to tak żeby macierz nie mogła być taka `[16, "Ola", 45.6]` bo chyba to nie powinno być dozwolone.